model/otel/registry.yaml

```diff
@@ -37,3 +41,73 @@ groups:
         brief: The version of the instrumentation scope - (`InstrumentationScope.Version` in OTLP).
         examples: ['1.0.0']
         stability: stable
+  - id: registry.otel.sdk
+    type: attribute_group
+    display_name: OTel SDK Telemetry Attributes
+    brief: Attributes used for OpenTelemetry SDK self-monitoring
+    attributes:
+      - id: otel.sdk.processor.type
+        type:
+          members:
+            - id: batching_span
+              value: "batch-span"
+              brief: >
+                The builtin SDK Batching Span Processor
+              stability: development
+            - id: simple_span
+              value: "simple-span"
+              brief: >
+                The builtin SDK Simple Span Processor
+              stability: development
+        stability: development
+        brief: >
+          A name identifying the type of the OpenTelemetry SDK processor.
+        examples: ["batch-span", "MyCustomProcessor"]
+      - id: otel.sdk.exporter.type
+        type:
+          members:
+            - id: otlp_grpc
+              value: "otlp-grpc"
+              brief: >
+                OTLP exporter over gRPC with protobuf serialization
+              stability: development
+            - id: otlp_http
+              value: "otlp-http"
+              brief: >
+                OTLP exporter over HTTP with protobuf serialization
+              stability: development
+            - id: otlp_http_json
+              value: "otlp-http-json"
+              brief: >
+                OTLP exporter over HTTP with JSON serialization
+              stability: development
+        stability: development
+        brief: >
+          A name identifying the type of the OpenTelemetry SDK exporter.
+        examples: ["otlp-grpc", "jaeger"]
+      - id: otel.sdk.component.name
+        type: string
+        stability: development
+        brief: >
+          A name uniquely identifying the instance of the OpenTelemetry SDK component within its containing SDK instance.
+        note: |
+          The attribute value MUST follow a `<type-name>/<instance-counter>` pattern, e.g. `batching_span_processor/0`.
```

Comment:
I just implemented this in https://github.com/open-telemetry/opentelemetry-go/pull/6153.  I would personally prefer if these were separated into separate attributes for the type and the instance

Reply:
I generally find it harder to use attributes that are joined in that way.  E.g. querying for `type=foo` is simpler than querying for `type=foo/.*`.  It can also be difficult to group by the type if it is joined with an ID. In PromQL, I would need to use `label_replace()` with a regex extract the prefix in order to group by it.

Generally, I don't expect the ID to be useful for users except as a way to distinguish different instantiations of the exporter/processor/sdk.  Given these components don't have a user-provided "name" or "Id" like the collector components do, it seems like it will generally be a UUID, or pointer, and won't be easy for users to consume.

Reply:
>  it will generally be a UUID, or pointer

It must not be of that kind, because that will blow up the cardinatlity considering application restarts. That's specifically why I prescribed the `isntance-counter` approach in the current PR state.

> I generally find it harder to use attributes that are joined in that way. E.g. querying for type=foo is simpler than querying for type=foo/.*. 

I understand. Then I would suggest to
 * Add `otel.sdk.component.type`
 * Keep `otel.sdk.component.name` but be less presriptive:
    * It MUST be a low-cardinality, unique within SDK instance identifier for the component.
    * It MAY be generated based on the `<type-name>/<instance-counter>` that is enforced in the current state of the PR
 
 WDYT? Also cc @jmacd

Reply:
Added `otel.sdk.component.type` in e737b7d.

Note that I've also changed the values for exporter to be specific to the kind of data: E.g. `otlp_http_exporter` changed to `otlp_http_span_exporter`. The reason is that the value needs to be unique for the component type, but exporters for logs, spans and metrics are different components, so `otlp_http_exporter` would have caused an overlap here.

Reply:
That doesn't seem necessary to me.  The metrics names are already split by signal type (i.e. contains "span"), so there is no need to split the exporter name by signal type.

Reply:
The reason why I thought that this is necessary is because due to my knowledge attributes should be designed to work even outside their current use-cases, because new, future use-cases may follow.

You are right, currently all metrics are split by signal type, but in the future there might be signal-independent use-cases. E.g. a `otel.sdk.component.cpu_usage` metric. Using just `otlp_http_exporter` instead of `otlp_http_span_exporter` would lead to ambiguities here.

That's why I think it is better to just accept the slightly longer, but more specific attribute values here.

Reply:
sounds ok to me